explainaboard/loaders/qa_open_domain.py

```diff
@@ -0,0 +1,53 @@
+from __future__ import annotations
+
+from explainaboard import TaskType
+from explainaboard.constants import FileType
+from explainaboard.loaders.file_loader import (
+    DatalabFileLoader,
+    FileLoader,
+    FileLoaderField,
+    JSONFileLoader,
+    TextFileLoader,
+)
+from explainaboard.loaders.loader import Loader
+from explainaboard.loaders.loader_registry import register_loader
+
+
+@register_loader(TaskType.qa_open_domain)
+class QAOpenDomainLoader(Loader):
+    @classmethod
+    def default_dataset_file_type(cls) -> FileType:
+        return FileType.json
+
+    @classmethod
+    def default_output_file_type(cls) -> FileType:
+        return FileType.text
+
+    @classmethod
+    def default_dataset_file_loaders(cls) -> dict[FileType, FileLoader]:
+        target_field_names = ["question", "answers", "question_types"]
+        return {
+            FileType.json: JSONFileLoader(
+                [
+                    FileLoaderField("question", target_field_names[0], str),
+                    FileLoaderField("answers", target_field_names[1], list),
+                    FileLoaderField("question_types", target_field_names[2], list),
```

Comment:
Do all open domain QA datasets have annotated question types? If not, then maybe this should be a custom feature of the dataset (I just added support for dataset-specific custom features to `explainaboard_web`), rather than something general for all open domain QA datasets.

Reply:
Aha, yes. I have almost the same plan.

Reply:
fixed in another pr: https://github.com/neulab/ExplainaBoard/pull/365